### PR TITLE
Tn21 get chats id emails

### PR DESCRIPTION
**Pull request by bvbps**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -2,3 +2,5 @@
 *.iml
 /.idea
 /target
+application.conf
```
Comment (IvoSaraiva) [inline .gitignore]: This is not supposed to be here.

File: app/repositories/slick/implementations/SlickChatsRepository.scala [hunk 2]
```diff
@@ -289,13 +289,29 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
   def moveChatToTrash(chatId: String, userId: String): Future[Boolean] =
     db.run(moveChatToTrashAction(chatId, userId))
 
+  private[implementations] def getEmailAction(chatId: String, emailId: String, userId: String) = {
+    for {
+      optionChat <- getChatAction(chatId, userId)
+    } yield optionChat match {
```
Comment (IvoSaraiva) [inline app/repositories/slick/implementations/SlickChatsRepository.scala]: This is a convoluted way of doing flatMap.
The for loop with just one line is unnecessary.

File: test/controllers/ChatControllerSpec.scala [hunk 3]
```diff
@@ -406,4 +406,40 @@ class ChatControllerSpec extends PlaySpec with Results with IdiomaticMockito {
...
+      val mockChatService = mock[ChatService]
+      mockChatService.getEmail(*, *, *)
+        .returns(Future.successful(Some(responseChatDto)))
+
+      val fakeAuthenticatedUserAction = new FakeAuthenticatedUserAction
+
+      val controller = new ChatController(cc, mockChatService, fakeAuthenticatedUserAction)
+
+      for {
+        result <- controller.getEmail(chatId, emailId).apply(FakeRequest())
```
Comment (IvoSaraiva) [inline test/controllers/ChatControllerSpec.scala]: Another for loop with just one entry.

File: .gitignore (hunk 1 above)
Comment (bvbps) [reply .gitignore]: Ups... Forgot to remove this

File: app/repositories/slick/implementations/SlickChatsRepository.scala (hunk 2 above)
Comment (bvbps) [reply app/repositories/slick/implementations/SlickChatsRepository.scala]: Okay, I'll change that

File: app/repositories/slick/implementations/SlickChatsRepository.scala [hunk 4]
```diff
@@ -289,13 +289,27 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
   def moveChatToTrash(chatId: String, userId: String): Future[Boolean] =
     db.run(moveChatToTrashAction(chatId, userId))
 
+  private[implementations] def getEmailAction(chatId: String, emailId: String, userId: String) = {
+    getChatAction(chatId, userId).map {
+      case Some(chat) =>
```
Comment (IvoSaraiva) [inline app/repositories/slick/implementations/SlickChatsRepository.scala]: This pattern match is just a map and filter of the Option.

File: test/repositories/slick/implementations/ChatsRepositorySpec.scala [hunk 5]
```diff
@@ -727,4 +727,69 @@ class ChatsRepositorySpec extends AsyncWordSpec with MustMatchers with OptionVal
...
+    }
+
+    "return None if the email is a draft of another user" in {
+      val draftEmailId = "fe4ff891-144a-4f61-af35-6d4a5ec76314"
+
+      //even if this user is a participant of the chat with "chatId", the user is not allowed to see drafts of other users
+      val notDraftOwnerUserId = "adcd6348-658a-4866-93c5-7e6d32271d8d"
+
+      for {
+        chat <- chatsRep.getEmail(chatId, draftEmailId, notDraftOwnerUserId)
```
Comment (IvoSaraiva) [inline test/repositories/slick/implementations/ChatsRepositorySpec.scala]: Still many for loops with just one entry.

**Pull request merged**